src/inc/til/rle.h

```diff
@@ -41,103 +41,79 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
 
             rle_iterator& operator++() noexcept
             {
-                operator+=(1);
+                ++_pos;
+                if (_pos == _it->length)
+                {
+                    ++_it;
+                    _pos = 0;
+                }
                 return *this;
             }
 
             rle_iterator operator++(int) noexcept
             {
-                rle_iterator tmp = *this;
-                operator+=(1);
+                auto tmp = *this;
+                ++tmp;
                 return tmp;
             }
 
             rle_iterator& operator--() noexcept
             {
-                operator-=(1);
+                if (_pos == 0)
+                {
+                    --_it;

```

Comment:
Will the vector [debug] iterator catch the out of bounds move if the user is at [0]?

Reply:
The `std::vector` one will. I'm not sure about the one in boost.
But this code is functionally equivalent to the `operator+=` implementation (just much simpler for inlining).